plugins/projectmgr/src/main/java/org/apache/ofbiz/project/Various.java

```diff
@@ -0,0 +1,110 @@
+                    GenericValue nextTask = assoc.getRelatedOne("ToWorkEffort", false);
+                    Timestamp newStartDate = task.getTimestamp("estimatedCompletionDate"); // start of next task the next day
+                    if (nextTask.get("estimatedStartDate") == null || nextTask.getTimestamp("estimatedStartDate").before(newStartDate)) {
+                        nextTask.put("estimatedStartDate", UtilDateTime.addDaysToTimestamp(task.getTimestamp("estimatedCompletionDate"), 1)); // start of next task the next day
+                        nextTask.put("estimatedCompletionDate", calculateCompletionDate(nextTask, task.getTimestamp("estimatedCompletionDate")));
+                        nextTask.store();
+                    }
+                    setDatesFollowingTasks(nextTask);
+                }
+            }
+
+        } catch (GenericEntityException e) {
+            Debug.logError("Could not updte task: " + e.getMessage(), module);
+        }
+    }
+
+    public static Timestamp calculateCompletionDate(GenericValue task, Timestamp startDate) {
+        Double plannedHours = 0.00;
+        try {
+            // get planned hours
+            List<GenericValue> standards = task.getRelated("WorkEffortSkillStandard", null, null, false);
+            for (GenericValue standard : standards) {
+                if (standard.getDouble("estimatedNumPeople") == null) {
+                    standard.put("estimatedNumPeople", Double.valueOf("1"));
+                }
+                if (standard.get("estimatedDuration") != null) {
+                    plannedHours += standard.getDouble("estimatedDuration") / standard.getDouble("estimatedNumPeople");
+                }
+            }
+
+        } catch (GenericEntityException e) {
+            Debug.logError("Could not updte task: " + e.getMessage(), module);
+        }
+        if (plannedHours == 0.00) {
+            plannedHours = Double.valueOf("24.00"); // default length of task is 3 days.
+        }
+
+        // only add days which are not saturday(7) or sunday(1)
+        int days = plannedHours.intValue() / 8;
+        while (days > 0) {
+            int dayNumber = UtilDateTime.dayNumber(startDate);
+            if (dayNumber != 1 && dayNumber != 7) {
+                days--;
+            }
+            startDate = UtilDateTime.addDaysToTimestamp(startDate, 1);
+        }
+        return startDate;
+    }
+
+    public static double calculateActualHours(Delegator delegator, String timesheetId) {
+        double actualHours = 0.00;
+        if (timesheetId != null) {
+            try {
+                List<GenericValue> actuals = EntityQuery.use(delegator).from("TimeEntry").where("timesheetId", timesheetId).queryList();
+                if (actuals.size() > 0) {
+                    for (GenericValue actual : actuals) {
+                        Double hour = (Double) actual.get("hours");
+                        double hours = hour;
```

Comment:
Rather weird Double handling in this file.

Reply:
agreed. java should unbox on assignment. fixed.